### PR TITLE
Azure affinity group / update new features in ping test

### PR DESCRIPTION
**Pull request by dq922**
Azure does not allow  ping through virtual ip address interface. The VM build process now attach a public IP to each VM and expose that as an ip_address

Added ping to allow across zones pinging with public IP with trigger: --ping_net_interface_type

### CODE REVIEW

**Comment by dq922** [comment]
I signed it!

File: perfkitbenchmarker/linux_benchmarks/ping_benchmark.py [hunk 1]
```diff
@@ -20,8 +20,18 @@
 import logging
 from perfkitbenchmarker import configs
 from perfkitbenchmarker import sample
+from perfkitbenchmarker import flags
+
 import re
 
+flags.DEFINE_enum('ping_net_interface_type', 'internal_ip',
```
Comment (cmccoy) [inline perfkitbenchmarker/linux_benchmarks/ping_benchmark.py]: Could you re-use the `--ip_addresses` flag defined in [`vm_util`](https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/blob/master/perfkitbenchmarker/vm_util.py#L95-L100)? We already have a lot of flags :)

File: perfkitbenchmarker/providers/azure/azure_virtual_machine.py [hunk 2]
```diff
@@ -198,9 +198,10 @@ def _Create(self):
     create_cmd = [AZURE_PATH,
                   'vm',
                   'create',
-                  '--affinity-group=%s' % self.network.affinity_group.name,
+                  '--location=%s' % self.zone,
                   '--virtual-network-name=%s' % self.network.vnet.name,
                   '--vm-size=%s' % self.machine_type,
+                  '--public-ip=%s' % self.name,
```
Comment (cmccoy) [inline perfkitbenchmarker/providers/azure/azure_virtual_machine.py]: Could you explain how this differs from the current method? Is the named public IP cleaned up when the instance is deleted?

**Comment by cmccoy** [comment]
@dq922 - thanks for this! I left a few notes.

Additionally, I looked at moving to location-based vnets in #778, but couldn't figure out how to ensure that a created VM and disks were associated with a given storage account without an affinity group. Do you have any ideas there?

**Comment by hildrum** [comment]
I'm creating release 1.3 soon; can this get pulled in for that?

File: perfkitbenchmarker/linux_benchmarks/ping_benchmark.py (hunk 1 above)
Comment (dq922) [inline perfkitbenchmarker/linux_benchmarks/ping_benchmark.py]: You're absolutely right, --ip_addresses param will cover the requirement to ping on the external interface. I'll remove the 'ping_net_interface_type' flags.

File: perfkitbenchmarker/providers/azure/azure_virtual_machine.py (hunk 2 above)
Comment (dq922) [inline perfkitbenchmarker/providers/azure/azure_virtual_machine.py]: Each VM in Azure comes automatically with a public and private IP address. The public (VIP) is not really associated with the VM but to the cloud service account that the VM lives in. By default, VIP cannot be pinged at all. 

If we do not utilize the public IP address, we would not be able to test latency across regions either on the VIP or private IP. Caveat is if we setup vnet-to-vnet VPN but it's out of scope of this test module. This contrasts with GCP where a subnet can be stretched across all of GPC's compute region. 

I'm proposing that we open up the ping test to allow cross-region ping test in addition to inter-datacenter ping.

**Comment by dq922** [comment]
@cmcocy Thanks for the feedback. I saw your #778 changes but am too confuse on how to associate the VM to storage account. 

I speculate that since MSFT flattened their storage / network architecture in ASM, and now all storage accounts / network accounts are directly associated with a region. IF we create a VM without specifying the storage account, I think it would just provision to the default storage account. Have to do further testing to validate.

It might be worthwhile to invest in developing support ARM mode where storage/network/compute resources are grouped in a resource group. It should address the majority of the issues see here.

**Comment by cmccoy** [comment]
> I speculate that since MSFT flattened their storage / network architecture in ASM, and now all storage accounts / network accounts are directly associated with a region. IF we create a VM without specifying the storage account, I think it would just provision to the default storage account. Have to do further testing to validate.

This is my concern - we create a standard or premium storage account for disk tests, and don't want to accidentally start always testing the user's default storage account type.

**Comment by voellm** [comment]
@hildrum  @cmccoy  - Whats the next step on the pull request?

**Comment by voellm** [comment]
@connormccoy - Are you able to complete this review?  It looks like you are working on it.  Thanks.

**Comment by cmccoy** [comment]
I believe this is still blocked on [this concern](https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/pull/833#issuecomment-190899074), namely that applying this change will break testing premium storage. @dq922, any ideas?

**Comment by noahl** [comment]
I believe this issue will be fixed by the move to Azure Resource Management in https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/pull/1112

**Comment by noahl** [comment]
I recently merged my ARM mode pull request, and we can now ping across zones. @dq922 , I'm going to close this, but please feel free to reopen if you want to add any more functionality on top of what is now in PKB. Contributions are always welcome! :)

**Pull request closed without merging**